pkg/kubernetes/pvc/v1alpha1/pvc.go

```diff
@@ -0,0 +1,74 @@
+package v1alpha1
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type predicate func(pvc) bool
+
+type pvc struct {
+	// actual pvc object
+	object *v1.PersistentVolumeClaim
+}
+
+type pvcList struct {
+	// list of pvc objects
+	items []pvc
+}
+
+type listBuilder struct {
+	list    *pvcList
+	filters []predicate
+}
+
+// ListBuilder enables building an instance
+// of pvcList
+func ListBuilder() *listBuilder {
+	return &listBuilder{list: &pvcList{}}
+}
+
+// HasName is filter function to filter pvc's
+// based on the name
+func HasName(name string) predicate {
+	return func(obj pvc) bool {
+		return strings.Contains(obj.object.GetName(), name)
+	}
+}
+
+// WithListObject builds the list of pvc instances
+// based on the provided pvc instances
+func (b *listBuilder) WithListObject(list *v1.PersistentVolumeClaimList) *listBuilder {
+	if list == nil {
+		return b
+	}
+	for i := range list.Items {
```

Comment:
How about below:
```go
  for _, p := range list.Items {
    b.list.items = append(b.list.items, pvc{&p}
  }
```

Reply:
it will not work.

Reply:
as the result will be the slice of last element.